ReSharper.FSharp/src/FSharp/FSharp.Psi/src/Impl/FSharpImplUtil.cs

```diff
@@ -171,6 +171,15 @@ public static string GetCompiledName([CanBeNull] this IIdentifier identifier)
     public static string GetSourceName([CanBeNull] this IIdentifier identifier) =>
       identifier?.Name ?? SharedImplUtil.MISSING_DECLARATION_NAME;
 
+    [NotNull]
+    public static string GetTypeParameterOrSourceName([CanBeNull] this IIdentifier identifier)
+    {
+      if (identifier == null) return SharedImplUtil.MISSING_DECLARATION_NAME;
+      return identifier is ITypeParameterId || identifier.PrevSibling?.GetTokenType() == FSharpTokenType.QUOTE
```

Comment:
Please don't check the token types of the siblings like this and fix the `ITypeParameterId` generation instead.

Reply:
removed